diff --git a/CMakeLists.txt b/CMakeLists.txt
index d2d8810..1ff278b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -346,13 +346,18 @@ if(UNIX)
 	install(FILES ${PC_FILE} DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
 endif()
 
+target_include_directories(yaml-cpp
+		PUBLIC
+			$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+#			$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/yaml-cpp>
+		)
 
 ###
 ### Extras
 ###
-if(YAML_CPP_BUILD_TESTS)
-	add_subdirectory(test)
-endif()
+#if(YAML_CPP_BUILD_TESTS)
+#	add_subdirectory(test)
+#endif()
 if(YAML_CPP_BUILD_TOOLS)
 	add_subdirectory(util)
 endif()
